### PR TITLE
Added test cases to invoke marshmallow field level validations.

### PR DESCRIPTION
**Pull request by inayathulla**
- Note: Validations are implemented already and while deserializing JSON it will check with schemas defined under graphql ->schemas and throws a validations error if there is any mismatch found.

# Description
Added test cases for T.OT

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Three test cases have been added.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] test_events.py
- [ ] test_assets.py
- [ ] test_plugins.py

**Test Configuration**:
* Python Version(s) Tested: 3.7

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/ot/test_assets.py [hunk 1]
```diff
@@ -11206,3 +11208,14 @@ def test_asset(fixture_ot):
 
     resp = fixture_ot.assets.asset("fff9f1b9-26db-4bf4-9d87-bd915aaede36")
     assert resp.next() == expected
+
+
+@pytest.mark.vcr()
+def test_assets_list_vcr(api):
```
Comment (aseemsavio) [inline tests/ot/test_assets.py:11214]: Would it be beneficial to additionally check if the asset is of type `Asset`?

File: tests/ot/test_assets.py [hunk 2]
```diff
@@ -11206,3 +11208,14 @@ def test_asset(fixture_ot):
...
+    """
+    Tests the assets Graphql list iterator with cassettes
+    """
+    assets = api.assets.list(limit=2)
+    assert isinstance(assets, OTGraphIterator)
+    asset = assets.next()
+    assert asset is not None
```
Comment (aseemsavio) [inline tests/ot/test_assets.py:11221]:
```suggestion
    assert asset is not None
    assert isinstance(asset, Asset)
```

File: tests/ot/test_events.py [hunk 3]
```diff
@@ -318,3 +320,14 @@ def test_list(fixture_ot):
...
+
+@pytest.mark.vcr()
+def test_events_list_vcr(api):
+    """
+    Tests the events Graphql list iterator with cassettes
+    """
+    events = api.events.list(sort=[{"field": "eventId", "direction": "DescNullLast"}], limit=2)
+    assert isinstance(events, OTGraphIterator)
+    event = events.next()
+    assert event is not None
```
Comment (aseemsavio) [inline tests/ot/test_events.py:333]:
```suggestion
    assert event is not None
    assert isinstance(event, Event)
```

File: tests/ot/test_plugins.py [hunk 4]
```diff
@@ -305,3 +307,14 @@ def test_plugin(fixture_ot):
...
+
+@pytest.mark.vcr()
+def test_plugins_list_vcr(api):
+    """
+    Tests the assets Graphql list iterator with cassettes
+    """
+    plugins = api.plugins.list(limit=2)
+    assert isinstance(plugins, OTGraphIterator)
+    plugin = plugins.next()
+    assert plugin is not None
```
Comment (aseemsavio) [inline tests/ot/test_plugins.py:320]:
```suggestion
    assert plugin is not None
    assert isinstance(plugin, Plugin)
```

File: tests/ot/test_assets.py (hunk 2 above)
Comment (inayathulla) [reply tests/ot/test_assets.py:11221]: It is taken care of by the assert isinstance(assets, OTGraphIterator), which evaluates and returns list of assets.

File: tests/ot/test_plugins.py (hunk 4 above)
Comment (inayathulla) [reply tests/ot/test_plugins.py:320]: same as above comment.

**Comment by aseemsavio** [comment]
lgtm

**Review by varap11 — APPROVED** [review]

**Pull request merged**